src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php

```diff
@@ -23,7 +37,19 @@ public function sign($input, $key, $password = null)
 
         $signature = null;
         openssl_sign($input, $signature, $keyResource, $this->getHashingAlgorithm());
+        
+        $asn1Decoder = new ASN1();
 
+        $asn1Decoded = $asn1Decoder->decodeBER($signature);
+        $asn1Decoded = $asn1Decoder->asn1map($asn1Decoded[0], self::$asn1Schema);
+        if( isset($asn1Decoded['r']) && isset($asn1Decoded['s']) &&
+            $asn1Decoded['r'] instanceof BigInteger              && 
+            $asn1Decoded['s'] instanceof BigInteger                 ) {
+
+            $signature = $asn1Decoded['r']->toBytes().$asn1Decoded['s']->toBytes();
```

Comment:
Yep, thanks. Are you currently working on a PR from your project? If so, I'd not waste my time fixing this.